bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -448,6 +448,7 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
                             ]
                             val oldThing = thingsMap.get(modelName).findFirst[it.UID == newThing.UID]
                             if (oldThing != null) {
+                                newThing.merge(oldThing)
```

Comment:
Me again :wink: 
I cannot have a look at the source code ATM but I assume the correctness depends how the merge method works.

I an "attribute" (label, channel, ...) of newThing and oldThing differs, who wins on a merge?
If the "attributes" of the thing to merge wins, then the new thing attributes are overwritten by the old ones...